src/crypto/openssl/cose_verifier.cpp

```diff
@@ -223,4 +228,89 @@ namespace ccf::crypto
   {
     return std::make_unique<COSEKeyVerifier_OpenSSL>(public_key);
   }
+
+  std::pair<std::string, std::string> extract_cose_endorsement_validity(
+    std::span<const uint8_t> cose_msg)
+  {
+    UsefulBufC msg{cose_msg.data(), cose_msg.size()};
+    QCBORError qcbor_result;
+
+    QCBORDecodeContext ctx;
+    QCBORDecode_Init(&ctx, msg, QCBOR_DECODE_MODE_NORMAL);
```

Comment:
Can init fail? Does it need a GetError too?

Reply:
It seems it can't
https://github.com/laurencelundblade/QCBOR/blob/master/inc/qcbor/qcbor_decode.h#L684

Also in the 3rd-arty provided tests I did see error handling after this one, but rather from the first iteration, whichever that is. In this case I'll add error handling to entering the bstr+map